org.lflang/src/org/lflang/validation/LFValidator.java

```diff
@@ -1304,15 +1299,88 @@ public void checkModeInstanceNamespace(Reactor reactor) {
             for (var mode : reactor.getModes()) {
                 for (var instantiation : mode.getInstantiations()) {
                     if (names.contains(instantiation.getName())) {
-                        error(String.format("Duplicate Instantiation '%s' in Reactor '%s'. (Instantiations are currently scoped on reactor level not modes)",
-                            instantiation.getName(), instantiation.getName()), instantiation, Literals.STATE_VAR__NAME);
+                        error(String.format("Duplicate Instantiation '%s'. (Instantiations are currently scoped on reactor level not modes)",
+                            instantiation.getName()), instantiation, Literals.INSTANTIATION__NAME);
                     }
                     names.add(instantiation.getName());
                 }
             }
         }
     }
 
+    @Check(CheckType.FAST)
+    public void checkMissingStateResetInMode(Reactor reactor) {
```

Comment:
It appears that a `reset` transition is the default, meaning that is no modifier is given in the transition effect, this is to mean that it is a reset transition. This could be confusing, because it may suggest that a "normal" transition is of a different kind than `reset` or `continue`. We had a similar problem with actions, and decided that it would be better to require a disambiguation from the programmer and thus only allow `logical action` or `physical action`, and not just `action`. I think we should take the same approach here, and enforce this in the validator so we can present a useful error message.